src/compiler/core.ts

```diff
@@ -714,20 +714,20 @@ namespace ts {
 
     export function fileExtensionIs(path: string, extension: string): boolean {
         let pathLen = path.length;
-        let extLen = extension.length;
-        return pathLen > extLen && path.substr(pathLen - extLen, extLen) === extension;
+        let extLen = extension.length + 1;
+        return pathLen > extLen && path.substr(pathLen - extLen, extLen) === "." + extension;
     }
 
     /**
      *  List of supported extensions in order of file resolution precedence.
      */
-    export const supportedExtensions = [".ts", ".tsx", ".d.ts"];
+    export const supportedTypeScriptExtensions = ["ts", "tsx", "d.ts"];
 
-    const extensionsToRemove = [".d.ts", ".ts", ".js", ".tsx", ".jsx"];
+    const extensionsToRemove = ["d.ts", "ts", "js", "tsx", "jsx"];

```

Comment:
we should not be using this any more, since users can specify additional jsExtenstions, this will fail unexpectedly.